connect/src/main/java/com/stripe/android/connect/EmbeddedComponentManager.kt

```diff
@@ -105,4 +132,46 @@ class EmbeddedComponentManager(
     data class Configuration(
         val publishableKey: String,
     ) : Parcelable
+
+    companion object {
+        private val launcherMap = mutableMapOf<Activity, ActivityResultLauncher<String>>()
+        private val permissionsFlow: MutableSharedFlow<Boolean> = MutableSharedFlow()
+
+        fun onActivityCreate(activity: ComponentActivity) {
+            val application = activity.application
+
+            application.registerActivityLifecycleCallbacks(object : ActivityLifecycleCallbacks {
+                override fun onActivityDestroyed(destroyedActivity: Activity) {
+                    // ensure we remove the activity and its launcher from our map, and unregister
+                    // this activity from future callbacks
+                    launcherMap.remove(destroyedActivity)
+                    if (destroyedActivity == activity) {
+                        application.unregisterActivityLifecycleCallbacks(this)
+                    }
+                }
+
+                override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) { /* no-op */ }
+
+                override fun onActivityStarted(activity: Activity) { /* no-op */ }
+
+                override fun onActivityResumed(activity: Activity) { /* no-op */ }
+
+                override fun onActivityPaused(activity: Activity) { /* no-op */ }
+
+                override fun onActivityStopped(activity: Activity) { /* no-op */ }
+
+                override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) { /* no-op */ }
+            })
+
+            activity.registerForActivityResult(
+                ActivityResultContracts.RequestPermission()
+            ) { isGranted ->
+                MainScope().launch {
+                    permissionsFlow.emit(isGranted)
+                }
+            }.also {
+                launcherMap[activity] = it
+            }
```

Comment:
Could simplify the assignment and maybe use `tryEmit` instead of launching. We might need to configure the `MutableSharedFlow` with `extraBufferCapacity = 1` though.
```suggestion
            launcherMap[activity] =
                activity.registerForActivityResult(
                    ActivityResultContracts.RequestPermission()
                ) { isGranted ->
                    permissionsFlow.tryEmit(isGranted)
                }
```

Reply:
I like this suggestion. Is there any risk of race conditions where multiple `tryEmit`s are called before the consumers from `permissionFlow`.first get to them? I think the risk is low and acceptable, curious on your thoughts @lng-stripe